compiler/rustc_const_eval/src/const_eval/error.rs

```diff
@@ -22,8 +22,22 @@ pub enum ConstEvalErrKind {
     ModifiedGlobal,
     RecursiveStatic,
     AssertFailure(AssertKind<ConstInt>),
-    Panic { msg: Symbol, line: u32, col: u32, file: Symbol },
+    Panic {
+        msg: Symbol,
+        line: u32,
+        col: u32,
+        file: Symbol,
+    },
     WriteThroughImmutablePointer,
+    /// Called `const_make_global` twice.
+    ConstMakeGlobalPtrAlreadyMadeGlobal(AllocId),
+    /// Called `const_make_global` on a non-heap pointer.
+    ConstMakeGlobalPtrIsNonHeap(String),
+    /// Called `const_make_global` on a dangling pointer.
+    ConstMakeGlobalWithDanglingPtr(String),
+    /// Called `const_make_global` on a pointer that does not start at the
+    /// beginning of an object.
+    ConstMakeGlobalWithOffset(String),
```

Comment:
Can you store a `Pointer` here instead of storing a string?

Reply:
It doesn't implement `IntoDiagArg` and other places pre-format it too.

Reply:
Usually we format in the diagnostic logic, see `compiler/rustc_const_eval/src/errors.rs`. Usually we do *not* pre-format, see 

https://github.com/rust-lang/rust/blob/c76d032f0144b650a438ee1efba89c475e0b115b/compiler/rustc_middle/src/mir/interpret/error.rs#L346-L429